druid/src/main/java/org/apache/calcite/adapter/druid/DruidSqlCastConverter.java

```diff
@@ -129,13 +129,13 @@
   private static String castCharToDateTime(
       TimeZone timeZone,
       String operand,
-      final SqlTypeName toType, String format) {
+      final SqlTypeName toType, final SqlTypeName fromType) {
```

Comment:
Though I can see the reason of this change,  to me it is more like a workaround. 

Now in the PR we call the method by code:
```java
castCharToDateTime(
          timeZone,
          castDateTimeToChar(DateTimeUtils.UTC_ZONE, operandExpression, fromType),
          toType,
          fromType);
```
I suggest to change to something like:
```java
castCharToDateTime(
          timeZone,
          castDateTimeToChar(DateTimeUtils.UTC_ZONE, operandExpression, fromType),
          toType,
          dateTimeFormatString(fromType));
```
This way the change of signature on `castCharToDateTime(...)`can be avoided.